nexus-repository-composer-it/src/test/java/org/sonatype/nexus/repository/composer/internal/ComposerProxyIT.java

```diff
@@ -81,16 +110,44 @@ public void unresponsiveRemoteProduces404() throws Exception {
     assertThat(status(proxyClient.get(BAD_PATH)), is(HttpStatus.NOT_FOUND));
   }
 
+  public void retrievePackagesJSONFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(FILE_PACKAGES)), is(HttpStatus.OK));
+
+    final Asset asset = findAsset(proxyRepo, FILE_PACKAGES);
+    assertThat(asset.name(), is(equalTo(FILE_PACKAGES)));
+    assertThat(asset.contentType(), is(equalTo(MIME_TYPE_JSON)));
+    assertThat(asset.format(), is(equalTo(FORMAT_NAME)));
+  }
+
+  public void retrieveListJSONFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(VALID_LIST_URL)), is(HttpStatus.OK));
+
+    final Asset asset = findAsset(proxyRepo, FILE_LIST);
+    assertThat(asset.name(), is(equalTo(FILE_LIST)));
+    assertThat(asset.contentType(), is(equalTo(MIME_TYPE_JSON)));
+    assertThat(asset.format(), is(equalTo(FORMAT_NAME)));
+  }
+
   @Test
-  public void retrieveProjectJSONFromProxyWhenRemoteOnline() throws Exception {
-    assertThat(status(proxyClient.get(VALID_PACKAGE_URL)), is(HttpStatus.OK));
+  public void retrieveProviderJSONFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(VALID_PROVIDER_URL)), is(HttpStatus.OK));
 
-    final Asset asset = findAsset(proxyRepo, VALID_PACKAGE_URL);
-    assertThat(asset.name(), is(equalTo(VALID_PACKAGE_URL)));
+    final Asset asset = findAsset(proxyRepo, VALID_PROVIDER_URL);
+    assertThat(asset.name(), is(equalTo(VALID_PROVIDER_URL)));
     assertThat(asset.contentType(), is(equalTo(MIME_TYPE_JSON)));
     assertThat(asset.format(), is(equalTo(FORMAT_NAME)));
   }
 
+  @Test
+  public void retrieveZipballFromProxyWhenRemoteOnline() throws Exception {
+    assertThat(status(proxyClient.get(VALID_ZIPBALL_URL)), is(HttpStatus.OK));
```

Comment:
Never heard of a ZIPBALL before, is this something I've just never heard of? Totally probable.

Reply:
haha for future eyes -- I got this form the matcher name:
![image](https://user-images.githubusercontent.com/10136383/66239551-02135f00-e6af-11e9-8f64-1e21e94b072e.png)